SPMeta2/SPMeta2.CSOM/ModelHandlers/WebModelHandler.cs

```diff
@@ -397,9 +398,22 @@ private static void MapProperties(Web web, WebDefinition webModel)
             if (webModel.IndexedPropertyKeys.Any())
             {
                 var props = web.AllProperties;
-                // TODO Not sure if property name should be hardcoded in here and if web.Update needs to be called here
-                props["vti_indexedpropertykeys"] = GetEncodedValueForSearchIndexProperty(webModel.IndexedPropertyKeys);
+               
+                // If other property keys have been set before, overwriting them would be bad
+                if (props["vti_indexedpropertykeys"] == null || props["vti_indexedpropertykeys"].ToString() == string.Empty)
+                {
+                    props["vti_indexedpropertykeys"] = GetEncodedValueForSearchIndexProperty(webModel.IndexedPropertyKeys);
+                }
+                else
+                {
+                    var indexPropertyValue = props["vti_indexedpropertykeys"].ToString();
+
+                    var indexList = GetDecodeValueForSearchIndexProperty(indexPropertyValue);
```

Comment:
These won't work well.

 indexList.AddRange(webModel.IndexedPropertyKeys) would add same-same props over with the several deployments. 

indexList.AddRange(webModel.IndexedPropertyKeys); should include case-insensitive Contains() check if the props is there or not.

props["vti_indexedpropertykeys"].ToString() == string.Empty ->
string.IsNUllOrEmpty9props["vti_indexedpropertykeys"].ToString()), there is also w ConvertUtils class that can help.

vti_indexedpropertykeys needs to be shared in some kind of internal M2 class, there was a class with some OOTB property names, we'll look up it.